bypass_antibot.py

```diff
@@ -5,143 +5,152 @@
 
 # Third Party Libraries
 import tweepy
+import feedparser
+
+# Local libraries
+from manage_rss import ManageRss
 
 
 class BypassAntiBot:
-    def __init__(self, api):
+    def __init__(self, api, flux_rss):
         """
         BypassAntiBot constructor containing anti bot avoidance logic
 
         :param api tweepy.API: api object for current tweepy user
+        :param flux_rss list: List of rss feeds
         """
         self.api = api
+        self.flux_rss = flux_rss
+        self.managerss = ManageRss()
 
     def bypass(self):
         """
         Randomly Retweet 5 tweets & Randomly Tweet 10 things.
         """
         try:
             logging.info("Bypass anti-bot protections ...")
-            self.randomretweet(5)
-            self.randomtweet(10)
+            self.randomretweet()
+            self.rss_and_tweet()
             logging.info("Anti-bot bypass completed !")
 
         except tweepy.TweepError as e:
             if e.api_code == 326:
                 pass
 
-    def remaining_retweet_calculation(self):
-        """ Check for number of ReTweets needed to bypass  """
-        retweet_count = 0
-        for tweet in self.api.user_timeline(count=200, tweet_mode="extended"):
-            if tweet.retweeted:
-                retweet_count += 1
-        remaining_retweet = round(-(40 - retweet_count/2) / (1/2))
-        return remaining_retweet
-
-    def remaining_retweetfollow_calculation(self):
-        """ Check for number of ReTweet Follow needed to bypass """
+    def random_retweet_calculation(self):
+        """
+        Check for number of ReTweet Follow needed to bypass
+        """
         follow_count, retweet_count = 0, 0
         for tweet in self.api.user_timeline(count=200, tweet_mode="extended"):
             if tweet.retweeted:
                 retweet_count += 1
                 if "FOLLOW" in tweet.full_text.upper():
                     follow_count += 1
-
-        remaining_tweet = (follow_count * 100) / retweet_count
-        return remaining_tweet
-
-    def randomretweet(self, retweet_count):
+        percent_RtFol = (follow_count * 100) / retweet_count
+        if percent_RtFol > 25:
+            randomrt = (follow_count * 4) - retweet_count
+            logging.info("They are " + str(round(percent_RtFol, 2)) + " % of RT related to giveaway, we will do " + str(
+                randomrt) + " RT random")
+            if randomrt > 15:
+                randomrt = 15
+                logging.info("We will only do " + str(randomrt) + " retweets for the moment.")
+        else:
+            logging.info("Il y a " + str(round(percent_RtFol, 2)) + " % de RT pour les concours on passe a la suite")
```

Comment:
```suggestion
            logging.info("There is " + str(round(percent_RtFol, 2)) + " % retweets for the giveaway, we move on to the next step.")
```

Histoire que ça colle au reste :)

Reply:
bien vu je l'ai loupé lui